ui/src/app/pages/data/data-set/export-data-set.tsx

```diff
@@ -0,0 +1,185 @@
+import {AnalyticsTracker} from 'app/utils/analytics';
+import {
+  DataSetRequest,
+  FileDetail,
+  KernelTypeEnum
+} from 'generated/fetch';
+import * as React from 'react';
+
+interface Props {
+  dataSetRequest: DataSetRequest;
+  newNotebook: Function;
+  notebookType?: Function;
+  updateNotebookName: Function;
+  workspaceNamespace: string;
+  workspaceFirecloudName: string;
+}
+
+interface State {
+  existingNotebooks: FileDetail[];
+  kernelType: KernelTypeEnum;
+  loading: boolean;
+  newNotebook: boolean;
+  notebookName: string;
+  notebooksLoading: boolean;
+  previewedKernelType: KernelTypeEnum;
+  queries: Map<KernelTypeEnum, String>;
+  seePreview: boolean;
+}
+const styles = {
+  codePreviewSelector: {
+    display: 'flex',
+    marginTop: '1rem'
+  },
+  codePreviewSelectorTab: {
+    width: '2.6rem'
+  }
+};
+
+export class ExportDataSet extends React.Component<Props, State> {
+
+  constructor(props: any) {
+    super(props);
+    this.state = {
+      existingNotebooks: [],
+      kernelType: KernelTypeEnum.Python,
+      loading: false,
+      newNotebook: true,
+      notebookName: '',
+      notebooksLoading: true,
+      previewedKernelType: KernelTypeEnum.Python,
+      queries: new Map([[KernelTypeEnum.Python, undefined], [KernelTypeEnum.R, undefined]]),
+      seePreview: false,
+    };
+  }
+
+  componentDidMount() {
+    this.loadNotebooks();
+    this.generateQuery();
+  }
+
+  private async loadNotebooks() {
+    try {
+      const {workspaceNamespace, workspaceFirecloudName} = this.props;
+      this.setState({notebooksLoading: true});
+      const existingNotebooks =
+          await workspacesApi().getNoteBookList(workspaceNamespace, workspaceFirecloudName);
+      this.setState({existingNotebooks});
+    } catch (error) {
+      console.error(error);
+    } finally {
+      this.setState({notebooksLoading: false});
+    }
+  }
+
+  async generateQuery() {
+    const {dataSetRequest, workspaceNamespace, workspaceFirecloudName} = this.props;
+    dataSetApi().generateCode(
+      workspaceNamespace,
+      workspaceFirecloudName,
+      KernelTypeEnum.Python.toString(),
+      dataSetRequest).then(pythonCode => {
+        this.setState(({queries}) => ({
+          queries: queries.set(KernelTypeEnum.Python, pythonCode.code)}));
+      });
+    dataSetApi().generateCode(
+      workspaceNamespace,
+      workspaceFirecloudName,
+      KernelTypeEnum.R.toString(),
+      dataSetRequest).then(rCode => {
+        this.setState(({queries}) => ({queries: queries.set(KernelTypeEnum.R, rCode.code)}));
+      });
+  }
+
+  setNotebookName(notebook) {
+    this.setState({notebookName: notebook});
+    this.props.newNotebook(true);
+    this.props.updateNotebookName(notebook);
+  }
+
+  setExistingNotebook(notebook) {
+    this.setState({newNotebook: notebook === ''});
+    this.setState({notebookName: notebook});
+    this.props.newNotebook(false);
```

Comment:
I think this should use `notebook === ''` instead of `false`. Currently, if I select an existing notebook from the list, then switch back to 'Create a new notebook' the 'EXPORT AND OPEN' button stays enabled. If clicked with an empty input, the export fails with 404 error.
<img width="524" alt="Screen Shot 2021-01-19 at 11 38 29 PM" src="https://user-images.githubusercontent.com/40036095/105131970-9fb2bd80-5aaf-11eb-8e71-4499db6d08ad.png">

Reply:
Fixed 

https://user-images.githubusercontent.com/34481816/105217386-cad5f500-5b21-11eb-9f6c-d3322bdaad64.mov